Bottom: 202830fc12da12cedef02f7c57e4148c2ac0a028
Top:    6d1278723784493d209b2f8ce3bdc61ac96772c2
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-29 11:17:51 -0500

Refresh of add-buildbot-ntel-worker

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index 91f0ea08..b68da51b 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -40,10 +40,12 @@
   tags:
     - buildbot
     - buildbot-worker
+    - buildbot-worker-ntel
   vars:
     asset_host: "{{hostvars['nexus']['uri']}}"
     asset_user: "buildbot"
     asset_password: "{{nexus_buildbot_password}}"
+    buildbot_master: "{{ hostvars[buildbot_hostname].ip_addr }}"
     container_etc_hosts:
       nexus.novatech-llc.com: "{{hostvars['nexus']['ip_addr']}}"
     known_hosts: "{{ gitlab_keys.stdout }}"
diff --git a/roles/buildbot-master/files/openembedded.py b/roles/buildbot-master/files/openembedded.py
index 042d283b..c193fc59 100755
--- a/roles/buildbot-master/files/openembedded.py
+++ b/roles/buildbot-master/files/openembedded.py
@@ -9,6 +9,7 @@ c = WorkerConfig = {}
 
 
 DEFAULT_BBFLAGS = '-k'
+DEFAULT_REPO = 'git@git.novatech-llc.com:ntel/setup-scripts.git'
 ASSET_HOST = os.getenv("ASSET_HOST", default="http://127.0.0.1")
 
 # Workers
@@ -16,11 +17,9 @@ ASSET_HOST = os.getenv("ASSET_HOST", default="http://127.0.0.1")
 # a Worker object, specifying a unique worker name and password.  The same
 # worker name and password must be configured on the worker.
 c['workers'] = [
-    worker.Worker("worker-ntel", "pass", max_builds=1),
+    worker.Worker("worker-ntel", "pass", max_builds=3),
 ]
 
-DEFAULT_REPO = 'git@git.novatech-llc.com:ntel/setup-scripts.git'
-
 # CHANGESOURCES
 c['change_source'] = [
     changes.GitPoller(
@@ -174,8 +173,7 @@ c['builders'] = []
 git_lock = util.MasterLock("git")
 
 
-@util.renderer
-def CurrentTime(props):
+def CurrentTime():
     from datetime import datetime
     import string
     dt = datetime.now()
@@ -188,13 +186,39 @@ def CurrentTime(props):
 def ComputeBuildProperties(props):
     newprops = {}
 
-    newprops['timestamp'] = ts = CurrentTime
+    newprops['timestamp'] = CurrentTime()
+
+    version = props.getProperty('version', default=newprops['timestamp'])
+    newprops['artifacts'] = {}
+    machines = [props.getProperty('machine')]
+
+    mc = props.getProperty('multiconfig')
+    if mc:
+        machines.extend(mc.split(' '))
+
+    for m in machines:
+        a = {}
 
-    newprops['dest'] = dest = util.Interpolate(
-        "/cache/images/%(prop:buildername)s")
+        a['dest'] = "/cache/images/%s" % (
+            props.getProperty('buildername')
+        )
 
-    newprops['archive'] = archive = util.Interpolate(
-        "%(kw:d)s/%(prop:machine)s.%(kw:t)s.tar.gz", d=dest, t=ts)
+        a['prefix'] = "%s-%s" % (
+            m,
+            version,
+        )
+
+        a['artifact'] = "%s.tar.gz" % (
+            a['prefix'],
+        )
+
+        a['url'] = "%s/%s/%s" % (
+            ASSET_HOST,
+            urlpath,
+            a['artifact'],
+        )
+
+        newprops['artifacts'][m] = a
 
     bbflags = props.getProperty('bbflags', DEFAULT_BBFLAGS)
     cache = props.getProperty('cache', True)
@@ -231,9 +255,7 @@ auto_conf = [
 class BitBakeConf(steps.StringDownload):
 
     def __init__(self, args, **kw):
-        lines = [
-        ]
-        lines.extend(args)
+        lines = list(args)
         configstring = '\n'.join(lines)
         conf_file = kw.setdefault('conf_file', 'auto.conf')
         sdkw = {
@@ -246,7 +268,8 @@ class BitBakeConf(steps.StringDownload):
 
 class BitBake(steps.Compile):
 
-    def __init__(self, package):
+    def __init__(self, package, **kwargs):
+
         kw = {
             'command': [
                 'bash',
@@ -258,41 +281,60 @@ class BitBake(steps.Compile):
             ],
             'description': 'building',
             'descriptionDone': 'build',
-            'descriptionSuffix': package,
-            'env': {'ENV': 'environment-ntel', 'BASH_ENV': 'environment-ntel'},
+            'env': {
+                'ENV': 'environment-ntel',
+                'BASH_ENV': 'environment-ntel',
+            },
             'flunkOnFailure': True,
             'haltOnFailure': False,
             'name': 'bitbake',
             'timeout': int(os.getenv('LONG_RUN_TIMEOUT', 600)),
             'warningPattern': "^WARNING: ",
         }
+        kw.update(kwargs)
         steps.Compile.__init__(self, **kw)
 
 
-class BitBakeArchive(steps.ShellCommand):
+class BitBakeArchive(steps.ShellSequence):
 
-    def __init__(self, **kw):
-        if 'machine' in kw:
-            self.machine = kw['machine']
-        else:
-            self.machine = util.Property('machine')
+    def __init__(self, **kwargs):
         kw = {
-            'command': ['ci-archive.sh',
-                        self.machine,
-                        util.Property("dest"),
-                        util.Property("timestamp")],
             'description': 'archiving',
             'descriptionDone': 'archive',
-            'env': {},
-            'flunkOnFailure': True,
-            'haltOnFailure': False,
-            'name': 'archive',
-            'env': {'PATH': ['/home/buildbot/', '${PATH}'],
+            'env': {
                     'ENV': 'environment-ntel',
                     'BASH_ENV': 'environment-ntel',
                     },
+            'flunkOnFailure': True,
+            'haltOnFailure': False,
+            'name': 'archive',
+            'timeout': int(os.getenv('LONG_RUN_TIMEOUT', 600)),
         }
-        steps.ShellCommand.__init__(self, **kw)
+        kw.update(kwargs)
+        kw = self.setupShellMixin(kw)
+        steps.ShellSequence.__init__(self, **kw)
+
+    def run(self):
+        commands = []
+
+        artifacts = self.getProperty('artifacts')
+        for m, art in artifacts.iteritems():
+            commands.append(util.ShellArg(
+                command="echo 'MACHINE=\"%s\"' >> conf/auto.conf" % (m),
+            ))
+
+            artfile = os.path.join(art['dest'], art['artifact'])
+            commands.append(util.ShellArg(
+                command=[
+                    'scripts/ci-archive.sh',
+                    art['prefix'],
+                    artfile
+                ],
+                logfile="Create %s archive" % (m),
+            ))
+        return steps.ShellSequence.runShellSequence(self, commands)
+
+
 
 
 class BitBakeFactory(util.BuildFactory):
@@ -301,14 +343,17 @@ class BitBakeFactory(util.BuildFactory):
         util.BuildFactory.__init__(self)
         self.addStep(steps.SetProperties(ComputeBuildProperties))
         self.addStep(steps.GitLab(
-            repourl=util.Property('repository'),
-            branch=util.Property('branch'),
-            codebase=util.Property('codebase'),
+            repourl=util.Property('repository', default=DEFAULT_REPO),
+            branch=util.Property('branch', default=DEFAULT_BRANCH),
+            codebase=DEFAULT_CODEBASE,
             mode=util.Interpolate("%(prop:clobber:#?|full|incremental)s"),
             method="clobber",
             locks=[git_lock.access('exclusive')],
             retry=(360, 5)))
         self.addStep(steps.ShellCommand(
+            name="configure",
+            description="configuring",
+            descriptionDone="configured",
             command=["./oebb.sh", "config", util.Property('machine')]
         ))
         self.addStep(BitBakeConf(auto_conf, conf_file='auto.conf'))
@@ -324,9 +369,9 @@ c['builders'].append(
         name="ntel-orionlxm",
         workernames=["worker-ntel"],
         factory=BitBakeFactory(
-            BitBake("orionlxm-swu-image"),
-            BitBake("orionlxm-disk-swu-image"),
-            BitBake("orion-headless-image -c populate_sdk"),
+            BitBake("orionlxm-swu-image", name="swu image"),
+            BitBake("orionlxm-disk-swu-image", name="disk swu image"),
+            BitBake("orion-headless-image -c populate_sdk", name="SDK"),
         ),
         properties={
             'machine': 'orionlxm',
@@ -340,9 +385,9 @@ c['builders'].append(
         name="ntel-orion-io",
         workernames=["worker-ntel"],
         factory=BitBakeFactory(
-            BitBake("-c cleanall u-boot-orion-io"),
-            BitBake("orion-io-swu-image"),
-            BitBake("orion-headless-image -c populate_sdk"),
+            BitBake("-c cleanall u-boot-orion-io", name="cleanup"),
+            BitBake("orion-io-swu-image", name="swu image"),
+            BitBake("orion-headless-image -c populate_sdk", name="SDK"),
         ),
         properties={
             'machine': 'orion-io',
@@ -356,10 +401,10 @@ c['builders'].append(
         name="ntel-orionlx-cpx",
         workernames=["worker-ntel"],
         factory=BitBakeFactory(
-            BitBake("-c cleanall gdk-pixbuf-native librsvg-native gtk-icon-utils-native"),
-            BitBake("orionlx-cpx-swu-image"),
-            BitBake("orionlx-cpx-disk-swu-image"),
-            BitBake("orion-graphical-image -c populate_sdk"),
+            BitBake("-c cleanall gdk-pixbuf-native librsvg-native gtk-icon-utils-native", name="cleanup"),
+            BitBake("orionlx-cpx-swu-image", name="swu image"),
+            BitBake("orionlx-cpx-disk-swu-image", name="disk swu image"),
+            BitBake("orion-graphical-image -c populate_sdk", name="SDK"),
         ),
         properties={
             'machine': 'orionlx-cpx',
@@ -373,9 +418,9 @@ c['builders'].append(
         name="ntel-orionlx-plus",
         workernames=["worker-ntel"],
         factory=BitBakeFactory(
-            BitBake("-c cleanall gdk-pixbuf-native librsvg-native gtk-icon-utils-native"),
-            BitBake("orionlx-plus-swu-image"),
-            BitBake("orion-graphical-image -c populate_sdk"),
+            BitBake("-c cleanall gdk-pixbuf-native librsvg-native gtk-icon-utils-native", name="cleanup"),
+            BitBake("orionlx-plus-swu-image", name="swu image"),
+            BitBake("orion-graphical-image -c populate_sdk", name="SDK"),
         ),
         properties={
             'machine': 'orionlx-plus',
@@ -389,9 +434,9 @@ c['builders'].append(
         name="ntel-qemux86-64",
         workernames=["worker-ntel"],
         factory=BitBakeFactory(
-            BitBake("gdk-pixbuf-native:do_cleanall"),
-            BitBake("orion-graphical-image"),
-            BitBake("orion-graphical-image -c populate_sdk"),
+            BitBake("gdk-pixbuf-native:do_cleanall", name="cleanup"),
+            BitBake("orion-graphical-image", name="image"),
+            BitBake("orion-graphical-image -c populate_sdk", name="SDK"),
         ),
         properties={
             'machine': 'qemux86-64',
@@ -408,15 +453,15 @@ c['builders'].append(
         factory=BitBakeFactory(
             BitBake(" gdk-pixbuf-native:do_cleanall"
                     " multiconfig:orionlx-cpx:gdk-pixbuf-native:do_cleanall"
-                    " multiconfig:orionlx-plus:gdk-pixbuf-native:do_cleanall"
-                    ),
+                    " multiconfig:orionlx-plus:gdk-pixbuf-native:do_cleanall",
+                    name="cleanup"),
             BitBake(" orion-graphical-image"
                     " multiconfig:orionlx-cpx:orionlx-cpx-swu-image"
                     " multiconfig:orionlx-cpx:orionlx-cpx-disk-swu-image"
                     " multiconfig:orionlx-plus:orionlx-plus-swu-image"
                     " multiconfig:orionlxm:orionlxm-swu-image"
-                    " multiconfig:orion-io:orion-io-swu-image"
-                    ),
+                    " multiconfig:orion-io:orion-io-swu-image",
+                    name="images"),
         ),
         properties={
             'machine': 'qemux86-64',
